examples/rcn.py

```diff
@@ -0,0 +1,392 @@
+        test_labels: Corresponding labels for the test images.
+    """
+
+    mnist_train_size = 60000
+    num_per_class = test_size // 10
+
+    print("Fetching the MNIST dataset")
+    dataset = fetch_openml("mnist_784", as_frame=False, cache=True)
+    print("Successfully downloaded the MNIST dataset")
+
+    mnist_images = dataset["data"]
+    mnist_labels = dataset["target"].astype("int")
+    full_mnist_test_images = mnist_images[mnist_train_size:]
+    full_mnist_test_labels = mnist_labels[mnist_train_size:]
+
+    np.random.seed(seed)
+
+    test_set = []
+    test_labels = []
+    for i in range(10):
+        idxs = np.random.choice(
+            np.argwhere(full_mnist_test_labels == i)[:, 0], num_per_class
+        )
+        for idx in idxs:
+            img = full_mnist_test_images[idx].reshape(28, 28)
+            img_arr = jax.image.resize(image=img, shape=(112, 112), method="bicubic")
+            img = jnp.pad(
+                img_arr,
+                pad_width=tuple([(p, p) for p in (44, 44)]),
+                mode="constant",
+                constant_values=0,
+            )
+
+            test_set.append(img)
+            test_labels.append(i)
+
+    return test_set, np.array(test_labels)
+
+
+# %%
+test_set, test_labels = fetch_mnist_dataset(test_size)
+train_labels = (
+    np.array([[i] * (train_size // 10) for i in range(10)]).reshape(1, -1).squeeze()
+)
+
+# %% [markdown]
+# # 2. Load the model
+
+# %%
+data = np.load(
+    f"example_data/rcn_{train_size}.npz", allow_pickle=True, encoding="latin1"
+)
+frcs, edges, suppression_masks, filters = (
+    data["frcs"][:train_size],
+    data["edges"][:train_size],
+    data["suppression_masks"],
+    data["filters"],
+)
+
+M = (2 * hps + 1) * (2 * vps + 1)
+
+# %% [markdown]
+# # 3. Visualize loaded model
+
+# %%
+img = np.ones((200, 200))
+pad = 55
+frc, edge = frcs[4], edges[4]
+fig, ax = plt.subplots(1, 1, figsize=(10, 10))
+for e in edge:
+    i1, i2, w = e
+    f1, r1, c1 = frc[i1]
+    f2, r2, c2 = frc[i2]
+
+    img[r1, c1] = 0
+    img[r2, c2] = 0
+    ax.text(
+        (c1 + c2) // 2 - pad, (r1 + r2) // 2 - pad, str(w), color="green", fontsize=15
+    )
+    ax.plot([c1 - pad, c2 - pad], [r1 - pad, r2 - pad], color="green", linewidth=0.5)
+
+ax.axis("off")
+ax.imshow(img[pad : 200 - pad, pad : 200 - pad], cmap="gray")
+
+
+# %% [markdown]
+# ## 3.1 Visualize the filters
+
+# %% [markdown]
+# The filters are used to detect the oriented edges on a given image. They are pre-computed using Gabor filters.
+
+# %%
+fig, ax = plt.subplots(4, 4, figsize=(10, 10))
+for i in range(filters.shape[0]):
+    idx = np.unravel_index(i, (4, 4))
+    ax[idx].imshow(filters[i], cmap="gray")
+    ax[idx].axis("off")
+
+fig.tight_layout()
+
+# %% [markdown]
+# # 4. Make pgmax graph
+
+# %% [markdown]
+# ## 4.1 Make variables
+
+# %%
+start = time.time()
+assert frcs.shape[0] == edges.shape[0]
+
```

Comment:
I would assign `M` here so that it is clear what it is and why it is defined as such (different categorical values PGM variables can take and how they correspond to pool choices)

Reply:
Added.